platform/settings.cpp

```diff
@@ -352,18 +352,78 @@ bool FromString<Transliteration::Mode>(string const & s, Transliteration::Mode &
   return true;
 }
 
-bool IsFirstLaunchForDate(int date)
-{
-  constexpr char const * kFirstLaunchKey = "FirstLaunchOnDate";
-  int savedDate;
-  if (!Get(kFirstLaunchKey, savedDate) || savedDate < date)
+UsageStats::UsageStats()
+  : m_firstLaunch("US_FirstLaunch")
+  , m_lastLaunch("US_LastLaunch")
+  , m_totalForeground("US_TotalForeground")
+  , m_sessions("US_SessionsCount")
+  , m_ss(StringStorage::Instance())
+{
+  std::string str;
+  uint64_t val;
+  if (m_ss.GetValue(m_totalForeground, str) && FromString(str, val))
+    m_totalForegroundTime = val;
+
+  if (m_ss.GetValue(m_sessions, str) && FromString(str, val))
+    m_sessionsCount = val;
+
+  if (!m_ss.GetValue(m_firstLaunch, str))
   {
-    Set(kFirstLaunchKey, date);
-    return true;
+    auto const fileTime = Platform::GetFileCreationDate(GetPlatform().SettingsPathForFile(SETTINGS_FILE_NAME));
+    if (fileTime > 0)
+    {
+      // Check that file wasn't created on this first launch (one minute threshold).
+      uint64_t const first = base::TimeTToSecondsSinceEpoch(fileTime);
```

Comment:
Может тогда сразу в платформе возвращать secondsSinceEpoch для удобства вместо time_t? И переименовать методы?

Reply:
Наверное лучше в platform оставить стандартный способ time_t.
Не вижу проблем с доп преобразованием.
AFAIK, time_t - это то еже самое что и seconds since epoch, просто нигде не специфицировано. Потому в релизе (по идее), тут не должно остаться ничего лишнего.

Reply:
Проблема в том, что на некоторых платформах time_t 32 битное.